### PR TITLE
Initiate a first pre-commit CI pass

### PR DESCRIPTION
**Pull request by kenodegard**
Instead of cluttering other benign PRs with the first pre-commit pass we do this.

### CODE REVIEW

**Comment by kenodegard** [comment]
pre-commit.ci autofix

**Comment by kenodegard** [comment]
@jezdez I know we are still experiencing broken tests on `conda-build`, are these failures of concern here?

File: setup.cfg [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [flake8]
 max-line-length = 180
-ignore = E122,E123,E126,E127,E128,E731,E722,W504,W605
+ignore = E122,E123,E126,E127,E128,E203,E731,E722,W503,W504,W605
```
Comment (jezdez) [inline setup.cfg:3]: Those ignore statements are a big worrying TBH, can we remove them?

**Comment by jezdez** [comment]
> @jezdez I know we are still experiencing broken tests on `conda-build`, are these failures of concern here?

It seems as if there is a dependency problem with urllib3 and brotlipy which makes sense to fix, I don't think they are triggered by this though.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[tool.black]
+line-length = 180
```
Comment (jezdez) [inline pyproject.toml:2]: Hmm, 180 seems way too much, is that how many characters some lines have?

**Review by jezdez — CHANGES_REQUESTED** [review]
I think this is a good first step but I'd rather have us strengthen the flake8 config a bit.

File: setup.cfg (hunk 1 above)
Comment (kenodegard) [reply setup.cfg:3]: I would love to remove most of these (eventually) but I think that would be out of scope for this PR?

File: pyproject.toml (hunk 2 above)
Comment (kenodegard) [reply pyproject.toml:2]: This was copied from the prior line-length set for flake8. I personally don't like conda or conda-build's line length policies and would much rather we switch to black's standard 88 line length. Would such a change be out of scope here too?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+target-version = ['py36', 'py37', 'py38']
+extend-exclude = '''
+^/(
+    conda/_vendor
+  | devenv
+)/
+'''
```
[the remark this answers is not in the document]
Comment (kenodegard) [reply pyproject.toml:9]: Good catch!

File: setup.cfg (hunk 1 above)
Comment (kenodegard) [reply setup.cfg:3]: As for the ignores I explicitly added (both are needed for black and flake8 to work together):
[E203](https://www.flake8rules.com/rules/E203.html): whitespace before ':'
[W503](https://www.flake8rules.com/rules/W503.html): line break occurred before a binary operator

Comment (jezdez) [reply setup.cfg:3]: Hmm, I'm wondering, mostly to better understand the impact, if we were to remove most of the ignores, would we see an immediate increase in code changes here in the PR?

File: pyproject.toml (hunk 2 above)
Comment (jezdez) [reply pyproject.toml:2]: I think this might be okay for this PR TBH since line length should only apply to a small number of lines in total.

Comment (kenodegard) [reply pyproject.toml:2]: True, do we want to use conda's line length or just use the default from black?

File: setup.cfg (hunk 1 above)
Comment (kenodegard) [reply setup.cfg:3]: No, but pre-commit CI's flake8 will probably blowup with complaints. Pre-commit CI by default runs pre-commit on all files, not just the changes, we can change this to be consistent with the local pre-commit behavior if we want.

these are changes that `black`/`darker` would address:
- [E122](https://www.flake8rules.com/rules/E122.html): under-indented continuation lines
- [E128](https://www.flake8rules.com/rules/E128.html): under-indented continuation lines
- [E126](https://www.flake8rules.com/rules/E126.html): over-indented continuation lines
- [E127](https://www.flake8rules.com/rules/E127.html): over-indented continuation lines
- [W503](https://www.flake8rules.com/rules/W503.html): line break occurred before a binary operator
- [W504](https://www.flake8rules.com/rules/W504.html): line break occurred after a binary operator
- [E123](https://www.flake8rules.com/rules/E123.html): align closing bracket with opening bracket's line
- [E203](https://www.flake8rules.com/rules/E203.html): whitespace before ':'

other changes we will need to manually fix:
- [E731](https://www.flake8rules.com/rules/E731.html): do not assign lambda, use def
- [E722](https://www.flake8rules.com/rules/E722.html): do not use bare except
- [W605](https://www.flake8rules.com/rules/W605.html): invalid escape sequence

File: pyproject.toml (hunk 2 above)
Comment (jezdez) [reply pyproject.toml:2]: Hm, I don't have a strong opinion other than preferring one standard than many. If both conda and conda-build seem unreasonable, we should default to black's default.

File: setup.cfg (hunk 1 above)
Comment (jezdez) [reply setup.cfg:3]: Oh interesting, I assumed the CI would also only just run against the changes, having consistency between dev and CI makes sense to me.

Comment (kenodegard) [reply setup.cfg:3]: Hm actually I'm not finding the option to change this (was very sure I'd seen it somewhere...). I believe pre-commit CI is effectively running `pre-commit run --all-files`. This has no impact on how `darker` works but means that `flake8` gets run on the entire code base. If we remove these undesired ignore statements flake8 will report an additional 700+ errors. Most will be relatively simple to address with the exception of E722 errors.

File: pyproject.toml (hunk 2 above)
Comment (kenodegard) [reply pyproject.toml]: Ok I went with black's default then, as black suggests we should also consider using [Bugbear's B950](https://github.com/PyCQA/flake8-bugbear#:~:text=tutorial%20for%20details.-,B950,-%3A%20Line%20too%20long) for linting.

**Review by jezdez — APPROVED** [review]
Thank you @kenodegard! 🚀

File: conda_build/launcher_sources/cpython-launcher-c-mods-for-setuptools.3.6.patch [hunk 4]
```diff
@@ -3,7 +3,7 @@
 @@ -8,11 +8,72 @@
   * Curt Hagenlocher (job management)
   */
- 
+
```
[the remark this answers is not in the document]
Comment (kenodegard) [reply conda_build/launcher_sources/cpython-launcher-c-mods-for-setuptools.3.6.patch:6]: Good point, will open PR for this.

[the remark this answers is not in the document]
Comment (mbargull) [reply conda_build/launcher_sources/cpython-launcher-c-mods-for-setuptools.3.6.patch:6]: Thanks!

**Pull request merged**